pkgs/servers/matrix-synapse/matrix-appservice-irc/default.nix

```diff
@@ -0,0 +1,29 @@
+{ pkgs, nodePackages, makeWrapper, nixosTests, nodejs, stdenv, lib, ... }:
+
+let
+
+  packageName = with lib; concatStrings (map (entry: (concatStrings (mapAttrsToList (key: value: "${key}-${value}") entry))) (importJSON ./package.json));
+
+  ourNodePackages = import ./node-composition.nix {
+    inherit pkgs nodejs;
+    inherit (stdenv.hostPlatform) system;
+  };
+in
+ourNodePackages."${packageName}".override {
+  nativeBuildInputs = [ makeWrapper ];
+  buildInputs = [ nodePackages.node-gyp-build ];
+
+  postInstall = ''
+    makeWrapper '${nodejs}/bin/node' "$out/bin/matrix-appservice-irc" \
+    --add-flags "$out/lib/node_modules/matrix-appservice-irc/app.js"
+  '';
+
+  meta = with lib; {
+    description = "Node.js IRC bridge for Matrix";
+    maintainers = with maintainers; [ puffnfresh jjjollyjim ];
+    homepage = "https://github.com/matrix-org/matrix-appservice-irc";
+    license = licenses.asl20;
+  };
+
+  passthru.tests.matrix-appservice-irc = nixosTests.matrix-appservice-irc;
```

Comment:
Let's also add this test to the matrix-synapse package and cc @Ma27 about it.

Reply:
I'm not sure if I like this: I'd like to see `matrix-synapse.tests` building even if this package breaks.